app/Filament/Concerns/SemesterSelectForOperator.php

```diff
@@ -0,0 +1,114 @@
+      Software as a Service (SaaS) by Canyon GBS LLC.
+    - Use of this software implies agreement to the license terms and conditions as stated
+      in the Elastic License 2.0.
+
+    For more information or inquiries please visit our website at
+    https://www.canyongbs.com or contact us via email at legal@canyongbs.com.
+
+</COPYRIGHT>
+*/
+
+namespace App\Filament\Concerns;
+
+use AdvisingApp\StudentDataModel\Models\Enrollment;
+use Filament\Forms\Components\Select;
+use Illuminate\Database\Eloquent\Builder;
+use Illuminate\Database\Eloquent\Model;
+
+trait SemesterSelectForOperator
+{
+    public static function semesterSelect(): Select
+    {
+        return Select::make('semesters')
+            ->label('Semester')
+            ->options(static::getSemesterOptions())
+            ->searchable()
+            ->multiple();
+    }
+
+    /**
+     * @return array<string, string>
+     */
+    public static function getSemesterOptions(): array
+    {
+        return Enrollment::query()
+            ->select('semester_name')
+            ->whereNotNull('semester_name')
+            ->distinct()
+            ->orderBy('semester_name')
+            ->get()
+            ->mapWithKeys(fn ($enrollment) => [$enrollment->semester_name => $enrollment->semester_name])
+            ->toArray();
+    }
+
+    /**
+     * @param  Builder<Model>  $query
+     *
+     * @return Builder<Model>
+     */
+    public function applyToBaseQuery(Builder $query): Builder
+    {
+        $relationshipName = $this->constraint->getRelationshipName();
+        $count = $this->settings['count'] ?? 1;
+        $semesters = $this->settings['semesters'] ?? null;
+
+        if (is_null($semesters)) {
+            $semesters = [];
+        } elseif (! is_array($semesters)) {
+            $semesters = [$semesters];
+        }
+
+        $semesters = array_values(array_filter($semesters, fn ($semester) => $semester !== null && $semester !== ''));
+        $lowerSemesters = array_map(fn ($semester) => mb_strtolower($semester), $semesters);
+
+        return $query->whereHas($relationshipName, function (Builder $query) use ($lowerSemesters) {
+            if (! empty($lowerSemesters)) {
+                $placeholders = implode(',', array_fill(0, count($lowerSemesters), '?'));
+                $query->whereRaw("LOWER(semester_name) IN ({$placeholders})", $lowerSemesters);
+            }
+        }, $this->getOperatorQuery(empty($lowerSemesters)), $count);
+    }
+
+    public function getSummary(): string
+    {
+        $summary = parent::getSummary();
+
+        if (! empty($this->settings['semesters'])) {
+            $semesters = $this->settings['semesters'];
+
+            if (! is_array($semesters)) {
+                $semesters = [$semesters];
+            }
+            $concatedSemester = implode(', ', $semesters);
+            $summary .= ' in semester "' . $concatedSemester . '"';
```

Comment:
I think it should be `semesters` when there is multiple and `semester` when there is only one